functions/src/handleAvailability.ts

```diff
@@ -0,0 +1,53 @@
+import { changeAvailability } from './adminUtils';
+import sendActionResponse from './sendActionResponse';
+
+const isAvailabilityText = (text: string) => {
+  return isMakeAvailable(text) || isMakeUnavailable(text);
+};
+
+const isMakeAvailable = (text: string) => {
+  return ['a', '-a', 'available'].includes(text);
+};
+
+const isMakeUnavailable = (text: string) => {
+  return ['u', '-u', 'unavailable'].includes(text);
+};
+
+const adjustAvailability = async (
+  text: string,
+  userName: string,
+  userId: string,
+  responseUrl: string
+) => {
+  if (isMakeAvailable(text)) {
```

Comment:
This entire section could be a little more DRY. For example, you could do a single call like `await changeAvailability(userId, isMakeAvailable(text))` and it would pass `true` or `false` correctly.